lib/matplotlib/figure.py

```diff
@@ -1144,6 +1145,33 @@ def subplots(self, nrows=1, ncols=1, sharex=False, sharey=False,
             # Returned axis array will be always 2-d, even if nrows=ncols=1.
             return axarr
 
+    def subplots_iterator(self, nrows=1, ncols=1, show_in_between=False,
+                          sharex=False, sharey=False, squeeze=True,
+                          subplot_kw=None, gridspec_kw=None):
+        """ Iteratively yields the axis object of a rows x cols subplot and
+        creates new subplots when needed"""
+        while True:
+            axes = self.subplots(
```

Comment:
This keeps putting the subplots back on the same `Figure`?  You could probably put this outside of the `while True` loop and just do a `ax.cla()` before yielding it.